extensions/ql-vscode/src/run-queries.ts

```diff
@@ -199,6 +209,30 @@ export class QueryEvaluationInfo {
     };
   }
 
+  /**
+   * Prints the end of the structured log summary file (starting from most expensive predicates)
+   * to the query server Output Window.
+   * @param qs The current QueryServerClient instance for this query.
+   */
+  printEvalLogSummary(qs: qsClient.QueryServerClient) {
+    if (this.hasEvalLogSummary()) {
+      fs.readFile(this.evalLogSummaryPath, (err, buffer) => {
+        if (err) {
+          throw new Error(`Could not read structured evaluator log summary file at ${this.evalLogSummaryPath}.`);
+        }
+        const summaryText = buffer.toString();
+        const mostExpensivePredicatesIndex = summaryText.indexOf('Most expensive predicates');
+
+        if (mostExpensivePredicatesIndex != -1) {
+          void qs.logger.log(' --- Evaluator Log Summary --- ', { additionalLogLocation: this.logPath });
+          void qs.logger.log(summaryText.substring(mostExpensivePredicatesIndex), { additionalLogLocation: this.logPath });
+        }
```

Comment:
Because there could be a variable number of "most expensive predicates" per stage, I couldn't think of a better way to parse out only this part of the logs for the output. cc @adityasharad because we discussed trying to avoid text parsing previously.

Reply:
Hmm, I am indeed a little uneasy about parsing the logs to do this. Perhaps a better option to achieve this without relying on parsing the logs could be to extend `codeql generate log-summary` with a flag that tells it to produce just this end-of-query summary (i.e. most expensive predicates and total evaluation times) without all the other information. Then, we could invoke that here to get the information without having to rely on log parsing. That gives us some robustness to changes to the log, and provides us with an isolated place we can potentially add more end-of-query information in future. It also avoids having to deal with the issue I mentioned above about consuming the file in a streaming manner.

Reply:
That makes sense to me. I am thinking of just extending the `--format` flag with a new option called `final` (or something along those lines) that only outputs the information we want in this window. We can workshop the option name 😄

Reply:
What does `codeql generate log-summary` currently print to stdout/stderr? Could we include the timing summary in that output, and just have the extension display that output in the Output window?

Reply:
I believe `generate log-summary` command does not print to stdout/stderr unless the output file for the summary is unspecified (in which case the entire summary text file is printed to stdout). It's possible that other values are printing to stdout/stderr that I haven't noticed in the codebase though 🤔 will defer to @edoardopirovano 

We were thinking of using a hidden option for `codeql generate log-summary` that only works with the `--format=text` option. Something like `codeql generate log-summary --format=text --end-of-query=${FILE}` It would write the end of query summaries to another file (which I guess is similar to writing it to stdout/stderr) and the Extension would consume that other file for the output window.

Reply:
Sounds reasonable; either of those will achieve what you want without requiring two command invocations.

Reply:
Ah yes, we came to that conclusion after I had implemented the `--format=final` option 🐢

Reply:
I did actually think of suggesting stdout for the end summary but I strayed away from it because it's already possible for the whole log to go to stdout so it seemed a bit strange to have that part of it always go there. Maybe stderr could be an option, though - I don't think we log anything there.

I think a separate file works too, though, and it has the advantage of leaving us stderr for something else in the future (maybe progress updates?).